src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -17,7 +17,7 @@ namespace System.IO.Compression
     public partial class ZipArchiveEntry
     {
         private ZipArchive _archive;
-        private readonly bool _originallyInArchive;
+        internal readonly bool _originallyInArchive;
```

Comment:
I think both these fields could be kept private by making `LocalHeaderOffsetComparer` a nested internal class under `ZipArchiveEntry`.  Nested classes have access to private members.

Reply:
`_originallyInArchive` and `_offsetOfLocalHeader` are used inside the `ZipArchive` class too, so I've exposed them as internal properties. I've also cleaned up the existing `_versionToExtract` field, which was marked internal and never used outside `ZipArchiveEntry`.